.azure/pipelines/ci.yml

```diff
@@ -29,7 +29,32 @@ variables:
 - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
   - name: _BuildArgs
     value: ''
+
 jobs:
+# Build Web.JS Linux x64
+- template: jobs/default-build.yml
```

Comment:
I might be late to the party, but I kinda question this. Why do we need a separate build step instead of letting the graph of project-references resolve this?

Reply:
Maybe I super don't understand what's going on here - but how is this OK? If it's possible for things to build in an unexpected order when you run `build.cmd` don't we **still** have a problem when building locally?

Reply:
@rynowak sorry I wasn't very clear. @JunTaoLuo is dealing w/ two separate issues
1. NodeJS projects don't reliably complete before C# projects are built
2. NodeJS isn't available on some of our platforms, (Think this was Windows ARM or Linux MUSL)

Since building the same things multiple times is wasteful anyway, I asked @JunTaoLuo to build the NodeJS projects once and use the artifacts in all the other jobs.

----
@JunTaoLuo that said, @rynowak's correct (if I'm reading everything correctly) that local builds may still fail due to the unreliable build ordering. I suggest changing the project references to `<MSBuild />` task invocations in a `BeforeBuild` target when transitioning from C# to NPM projects. It's ugly but should be more reliable.

Reply:
No, there shouldn't be any ordering issue on local builds after this change. The reason why we needed to add this build step, as @dougbu elaborated, is that we don't have a NodeJS installation on Linux Musl/ Linus Musl x64. It's less for optimizing for resource use.